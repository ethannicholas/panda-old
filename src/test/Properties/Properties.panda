class Properties {
    property p1:String
    property p2:Int

    @protected
    function get_p1():String {
        return "421"
    }

    @private
    method set_p2(p2:Int) {
        Console.writeLine(p2)
    }
}

class Sub : Properties {
    @override
    function get_p1():String {
        return "872"
    }
}

def a := Properties()
a.p2 := a.p1.convert()

def b := Sub()
b.p2 := b.p1.convert()