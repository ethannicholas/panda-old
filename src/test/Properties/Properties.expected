setting name to: test
testOVERRIDDEN
success
